File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        """
        time: O(n)
        space: O(n)
        """
        res = []
        for num in nums:
            if num != val:
                res.append(num)

        for i in range(len(res)):
            nums[i] = res[i]

        return len(res)

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        List<Integer> uniqueNums = new ArrayList<>();

        /**
        Time: O(n)
        Space: O(n)
        */
        for (int num: nums) {
            if (uniqueNums.contains(num)) {
                return true;
            }
            uniqueNums.add(num);
        }

        return false;
    }
}